File: backend/app/models/agent.py
from pydantic import BaseModel


class SectionEditRequest(BaseModel):
    section_id: str
    instruction: str


class DiffActionRequest(BaseModel):
    section_ids: list[str]


class StagedEditSummary(BaseModel):
    section_id: str
    old_latex: str
    new_latex: str
    rationale: str

File: frontend/src/types/agent.ts
export type AgentEvent =
  | { type: "tool_call"; name: string; arguments: Record<string, unknown> }
  | { type: "tool_result"; name: string; result: unknown }
  | { type: "section_proposed"; section_id: string }
  | { type: "validation_error"; errors: string[] }
  | { type: "done"; message: string }
  | { type: "error"; message: string };

export interface StagedEditSummary {
  section_id: string;
  old_latex: string;
  new_latex: string;
  rationale: string;
}
